src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -188,6 +187,11 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
             IEnumerable<PackageReference> calculatedTransitivePackages = Enumerable.Empty<PackageReference>();
             if (includeTransitivePackages)
             {
+                if (targetsList == null)
+                {
+                    targetsList = await GetTargetsListAsync(assetsFilePath, token);
```

Comment:
`targetsList` was only calculated when `IsComputationNeeded` was true and we need to update the list when getting transitives